modules/product/src/ProductTestTrait.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\commerce_product;
+
+use Drupal\commerce_product\Entity\ProductAttribute;
+use Drupal\commerce_product\Entity\ProductAttributeValue;
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Provides methods for working with products, variations, and attributes.
+ *
+ * This trait is meant to be used only by test classes.
+ *
+ * @todo Move to \Drupal\Tests\commerce_product post-SimpleTest.
+ */
+trait ProductTestTrait {
+
+  /**
+   * Creates a set of product cariations.
+   *
+   * The $values_matrix parameter contains the price and attributes information.
+   *
+   * Example:
+   * [
+   *   [
+   *      'price' => 9.99
+   *      'attribute_color' => 2,
+   *      'attribute_size' => 10,
+   *   ],
+   *   [
+   *      'price' => 9.99
+   *      'attribute_color' => 2,
+   *      'attribute_size' => 6,
+   *   ],
+   * ]
+   *
+   * @param string $type
+   *   The variation type.
+   * @param array $values_matrix
+   *   Array of values.
+   * @param string $currency_code
+   *   The currency code.
+   *
+   * @return \Drupal\commerce_product\Entity\ProductVariationInterface[]
+   *   Array of created variations
+   */
+  protected function createProductVariations($type = 'default', $values_matrix = [], $currency_code = 'USD') {
```

Comment:
Feels odd to have $currency_code, this belongs in $values_matrix.

It's inconsistent that we have createStore(), but createProductVariations(). Let's do createProductVariation() and the implementing tests can do a foreach.